openedx/features/enterprise_support/api.py

```diff
@@ -43,6 +41,39 @@ class EnterpriseApiException(Exception):
     pass
 
 
+class ConsentApiClient(object):
+
+    def __init__(self):
+        self.user = User.objects.get(username=settings.ENTERPRISE_SERVICE_WORKER_USERNAME)
+        jwt = JwtBuilder(self.user).build_token([])
+        self.client = EdxRestApiClient(
+            configuration_helpers.get_value('CONSENT_API_URL', settings.CONSENT_API_URL),
+            jwt=jwt,
+            append_slash=False,
+        )
+        self.consent_endpoint = self.client.data_sharing_consent
+
+    def revoke_consent(self, **kwargs):
+        return self.consent_endpoint.delete(**kwargs)
+
+    def provide_consent(self, **kwargs):
+        return self.consent_endpoint.post(kwargs)
+
+    def consent_needed(self, enrollment_exists=False, **kwargs):
+        from slumber.exceptions import HttpNotFoundError
+        try:
+            return self.consent_endpoint.get(**kwargs)['consent_required']
+        except HttpNotFoundError as e:
```

Comment:
@haikuginger I think it would be worth explaining in a comment that `HttpNotFoundError` is what we expect if no consent record exists for the current learner/EC/course.